Client/Frontend/Browser/TopTabsViewController.swift

```diff
@@ -217,13 +222,26 @@ class TopTabsViewController: UIViewController {
             }
         }
     }
-
+    
+    private func handleFadeOutOfTabs() {
+        guard let currentTab = tabManager.selectedTab, let index = topTabDisplayManager.dataStore.index(of: currentTab), !collectionView.frame.isEmpty else {
+            return
+        }
+        
+        // Check wether first or last tab is being selected
+        if (index == 0 || index == topTabDisplayManager.dataStore.count - 1) {
+            topTabFader.isFaderActive(false)
+        } else {
+            topTabFader.isFaderActive(true)
+        }
+    }
 }
 
 extension TopTabsViewController: TabDisplayer {
 
     func focusSelectedTab() {
         self.scrollToCurrentTab(true)
+        self.handleFadeOutOfTabs()
```

Comment:
So right now fading will be disabled or enabled only when we select a tab. This opens the door for the following edge case:
1. I have a lot of tabs in the top tab tray
2. I select the first tab but then scroll to see more tabs to the left (first tab is disappearing out of the view, but I still have it selected)
3. This result in having no fading on any of the cells:
![Screen Shot 2022-01-27 at 6 19 27 PM](https://user-images.githubusercontent.com/11338480/151459844-caa79e74-8053-4301-bbff-51d12ebd74fc.png)

Where is should be instead faded:
![Screen Shot 2022-01-27 at 6 22 45 PM](https://user-images.githubusercontent.com/11338480/151459847-815be2d7-828c-4885-81ba-13c9035eb9c1.png)

There's a need to check for that case as well.

Reply:
I understand the problem. To solve it, it would be probably necessary to use `UICollectionViewDelegate` to detect, wether the user has scrolled to the first tab or the last tab. Only when the scroll position is at the most left or right, then the fade out for this site should be disabled.
Unfortunately I cannot find the right place to conform to `UICollectionViewDelegate` and trigger an appropriate function in `TopTabsViewController`.

Reply:
One way to solve the delegate problem is to add a delegate to `TopTabsLayoutDelegate` which I'll call `scrollViewDelegate` for now. Then in `TopTabsViewController` we can set this delegate to self. 

Here's some code example of what this could looks like: https://github.com/finebel/firefox-ios/compare/v97.0...lmarceau:lm/scrollExample?expand=1
Let me know if this helps!

Reply:
Thanks. This was super helpful!

Reply:
This is the behaviour now:

https://user-images.githubusercontent.com/46824694/151610969-b15fb06e-c6b6-445a-b5a2-a72a0123656b.mov

: